### ISSUE TITLE
`gather` drop-in replacement

### ISSUE THREAD

**Issue by prophile** [florimondmanca/aiometer#41]
Hello!

asyncio's own `gather` is a bit of a dated design but still pretty common, at least in codebases I'm dealing with at the moment—converting them to `amap` would be nice but it would also be dead useful to be able to drop-in replace with an `aiometer.gather` in a like-for-like fashion to add metering separately.

It's not hard to shim on top of amap but would it be something you'd consider bringing into scope in the library itself?

**Comment by florimondmanca** [comment]
Hi @prophile 

This is interesting at first, but I’m not sure.

Like you said, gather has its own quirks. For example, I believe it’s possible to pass a flag that changes its behavior quite substantially.

There’s already run_all and run_any helpers on aiometer which build on top of amap. Don’t those do the trick?

**Comment by prophile** [comment]
Potentially yes, and switching over to them would be an aspiration. But changing something from using `asyncio.gather` to `aiometer.run_all` isn't necessarily drop-in in the sense that things might rely on `asyncio.gather`'s peculiarities and changing them might cause a breakage.

For me (us) I think that's useful to do, but it would also be useful to get metering in the short term before doing the bigger job of changing over to `run_all`.

**Comment by florimondmanca** [comment]
I _think_ I understand where you’re at, but let me rephrase this.

What piece of public API do you think we could add to aiometer that would address this use case? If we’d add a gather « like for like but with metering », what would be its signature and a few code usage examples?
